react/containers/ChannelSelect/view.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import ChannelLoginForm from 'containers/ChannelLoginForm';
+import ChannelCreateForm from 'containers/ChannelCreateForm';
+
+const LOGIN = 'Existing';
+const CREATE = 'New';
+
+class ChannelSelect extends React.Component {
+  constructor (props) {
+    super(props);
+    this.state = {
+      selectedOption: LOGIN,
+    };
+    this.toggleAnonymousPublish = this.toggleAnonymousPublish.bind(this);
+    this.handleSelection = this.handleSelection.bind(this);
+    this.selectOption = this.selectOption.bind(this);
+  }
+  componentWillMount () {
+    console.log('ChannelSelector will mount');
+    if (this.props.loggedInChannelName) {
+      this.selectOption(this.props.loggedInChannelName);
+    }
+  }
+  componentWillReceiveProps ({ loggedInChannelName }) {
+    if (loggedInChannelName) {
+      this.selectOption(loggedInChannelName);
```

Comment:
Instead of reading prop values and adding them to state, you should just use the prop values and not worry about internal state.

I try to avoid internal state whenever possible, one exception is the app homepage, where we store if the user can scroll left or right in a card row. But that isn't dependent on any prop value, just by what is currently on the screen.

Whenever you separate the data into two sources it can cause some weird issues.

Reply:
The issue I am struggling with, is that I (think) I need the `<select>` element to be a controlled component, and if so then it seems like it would be best to use internal state rather than the store, since no other component should care what option is selected unless this component decides to update the state of the store.  The reason I am setting the state when the component mounts and when it receives props, is that if another component updates the store when a new channel is logged into, then I want this `<select>` element to re-render with that option selected.  Any ideas on how I can approach this differently so that I can eliminate internal state while still accomplishing those goals?